agent/proxycfg/upstreams.go

```diff
@@ -479,8 +479,8 @@ func (s *handlerUpstreams) watchUpstreamTarget(ctx context.Context, snap *Config
 	var entMeta acl.EnterpriseMeta
 	entMeta.Merge(opts.entMeta)
 
-	ctx, cancel := context.WithCancel(ctx)
-	err := s.dataSources.Health.Notify(ctx, &structs.ServiceSpecificRequest{
+	peerCtx, cancel := context.WithCancel(ctx)
```

Comment:
I see several `ctx` renames but wouldn't expect an overwriting issue with the way they were since they're always assigned inside a new scope. What motivated this change?

Reply:
ahhh so the innermost one actually needs to be a child of the outermost one and not its enclosing loop?

Reply:
yes, we need both watches data to be able to generate the right config. If we go with the chain then every time the one in the middle is canceled we loose the data of the innermost one because it get canceled with it.